runtime/src/lib.rs

```diff
@@ -91,10 +95,15 @@ pub use content::MaxNumber;
 
 /// This runtime version.
 pub const VERSION: RuntimeVersion = RuntimeVersion {
+    #[cfg(not(any(feature = "staging_runtime", feature = "testing_runtime")))]
     spec_name: create_runtime_str!("joystream-node"),
+    #[cfg(feature = "staging_runtime")]
+    spec_name: create_runtime_str!("joystream-node-staging"),
+    #[cfg(feature = "testing_runtime")]
+    spec_name: create_runtime_str!("joystream-node-testing"),
```

Comment:
Fair point, I considered modifying the runtime authoring version instead of the spec name.
The only reason I wanted to make this change was prevent mistake in doing a runtime upgrade with a wasm built for a different profile. 

> Looks good, I tested some configurations - they work. However, the bencmarking failed (at least the previous run).
> 
> I have a couple of comments:
> 
>     * Did you consider placing all the feature-gated code in a single file to simplify future configuration changes?
> 

I did but I wanted to keep the refactoring of the main lib.rs to a minimum although I had proposed to do so in the intial issue. Perhaps this can be don in a future PR.

>     * `cargo build --release --features staging_runtime,testing_runtime` this command fails. It's possible to avoid it properly but I'm not sure it worth the trouble.

Yes it fails with the compiler error at the top of lib.rs as they are mutually exclusive features. Is there a better way?

Reply:
Removed any change in runtime version based on runtime profile.